### ISSUE TITLE
V3: Mark non-consumer Zowe CLI SDK APIs as @internal

### ISSUE THREAD

**Issue by gejohnston** [zowe/zowe-cli#2083]
**Is your feature or enhancement request related to a problem or limitation? Please describe**

Many APIs are public so that they can be used by other SDK classes. However, once public, those APIs cannot be modified without introducing a breaking change. This can cause significant technical debt to build up leading into a new major Zowe release, such as postponed modifications, duplicated functions, deprecation and later removal.

**Describe your enhancement idea**

Review the SDK APIs and mark functions that are not intended to be used by external consumers as @internal. Such modifications will be a breaking change, so this work would have to be done before the release of Zowe V3.

If consumers request that an @internal function be made externally available, that action could be taken at any later time, since that action would **NOT** be a breaking change.

**Describe alternatives you've considered**

**Provide any additional context**

Marked as a medium priority since it either must be done before Zowe V3 or postponed for years.

**Comment by awharn** [comment]
Completed as part of #2159
